### ISSUE TITLE
Multi-node failure on optimize_spmd10.rg

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/legion#246]
Command to repro (note, uses: `-fflow 0 -finner 0 -fleaf 0`):

```
./install.py --debug --gasnet
i=0; while mpirun -n 2 -H localhost --bind-to none -x REALM_SYNTHETIC_CORE_MAP= -x INCLUDE_PATH -x LD_LIBRARY_PATH -x TERRA_PATH ./regent.py ./tests/regent/run_pass/optimize_spmd10.rg -ll:cpu 2 -fflow 0 -finner 0 -fleaf 0; do let i++; echo $i; done
```

Bug is non-deterministic, repro rate appears to be between 1 in 10 to 1 in 100. I recommend running the above command in parallel in 3 or 4 separate shells if you're trying to repro on sapling.

I do *not* appear to be able to repro on a single node, with or without the default_mapper.cc:1996 hack.

It *does* repro in Legion Spy! Log files:

  * [spy1_0.txt](https://github.com/StanfordLegion/legion/files/974385/spy1_0.txt)
  * [spy1_1.txt](https://github.com/StanfordLegion/legion/files/974386/spy1_1.txt)

And this produces a Legion Spy error:

```
ERROR: avg_ac (UID=16) failed to generate a close operation for field a (101) of region requirement 2 at q (4,2,2) for normal close operation
```

**Comment by lightsighter** [comment]
This error message from Legion Spy is actually incorrect because not all of the logging information made it into the file about the partitions and which points were contained in each region.  By modifying the log files to include the important information about which points are in which subregions, we get a different error message from Legion Spy:

`ERROR: Missing source precondition for Realm Copy (34) on field b (102) for op region requirement 1 of inc_ba on Mapping Op 6`

I now think this issue is the same as issue #245.

**Comment by lightsighter** [comment]
Actually if you run Legion Spy on the log files from a multi-node run with the above configuration it will fail deterministically regardless of whether you get the right answer or not.  I should be able to track this one down pretty easily.

**Comment by lightsighter** [comment]
Managed to fix one bug so now Legion Spy only fails its verification some of the time. :)  Still a second bug running around here.  I can see it, but I'm going to wait until the morning to fix it so that I will hopefully be more awake.

**Comment by lightsighter** [comment]
This should be fixed with 96841fe. Assigning back to @elliottslaughter to confirm.

**Comment by elliottslaughter** [comment]
Confirmed. I ran 1300 tests on my own, and gitlab agrees.
